frontend/src/components/Admin/TeamEvent/TeamEventDashboard.tsx

```diff
@@ -51,11 +52,58 @@ const TeamEventDashboard: React.FC = () => {
 
   if (isLoading) return <Loader active>Fetching team event data...</Loader>;
 
+  const handleExportToCsv = () => {
+    const csvData = allMembers.map((member) => {
+      const totalCredits = teamEvents.reduce(
+        (val, event) => val + calculateTotalCreditsForEvent(member, event),
+        0
+      );
+      const initiativeCredits = teamEvents.reduce(
+        (val, event) => val + calculateInitiativeCreditsForEvent(member, event),
+        0
+      );
+
+      const data = teamEvents.reduce(
+        (prev, event) => ({
+          ...prev,
+          [event.name]: calculateTotalCreditsForEvent(member, event)
+        }),
+        {
+          Name: `${member.firstName} ${member.lastName}`,
+          NetID: `${member.netid}`,
+          Total: totalCredits,
+          Initiative: initiativeCredits
+        }
+      );
+
+      return data;
+    });
+
+    const options: Options = {
+      filename: `TEC_Dashboard`,
+      fieldSeparator: ',',
+      quoteStrings: '"',
+      decimalSeparator: '.',
+      showLabels: true,
+      showTitle: true,
+      title: `TEC Dashboard`,
+      useTextFile: false,
+      useBom: true,
+      useKeysAsHeaders: true
+    };
+
+    const csvExporter = new ExportToCsv(options);
+    csvExporter.generateCsv(csvData);
+  };
+
   return (
     <div className={styles.dashboardContainer}>
-      <Header as="h1" textAlign="center">
-        Team Event Dashboard
-      </Header>
+      <div className={styles.headerContainer}>
+        <Header as="h1">Team Event Dashboard</Header>
+        <div className={styles.csvButton}>
+          <Button onClick={() => handleExportToCsv()}>Export to CSV</Button>
```

Comment:
nit: the arrow function can just be replaced with `handleExportToCsv`. (also, henry mentioned in my PR that using anonymous functions as inputs is frowned upon since its less performant.)

Reply:
fyi the performance decrease is _extremely marginal_, we're talking like a couple milliseconds maybe. If there's a lot of these and _a lot of re-renders_ (like _A LOT_) ok sure maybe there's noticeable performance issues but for the most part it's nbd. 

but mostly for readability it's a little nicer to use a pre-defined handler function. that func could potentially also be re-used across different handlers